### PR TITLE
Update gmt4 version

### PR DESCRIPTION
**Pull request by cll12**
### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [ ] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [ ] Built your formula locally prior to submission with `brew install <formula>`?

### New formula: have you

- [ ] Written a sensible test? (The best test is to compile and run a sample program.)

### Updates to existing formula: have you

- [ ] Removed the `revision` line, if any, when bumping the version number?
- [ ] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [ ] Not altered the `bottle` section?
- [ ] Checked that the tests still pass?

There is a new GMT4 version as of Oct 1.

### CODE REVIEW

File: gmt4.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class Gmt4 < Formula
   homepage "http://gmt.soest.hawaii.edu/"
-  url "ftp://ftp.soest.hawaii.edu/gmt/gmt-4.5.14-src.tar.bz2"
```
Comment (cll12) [inline gmt4.rb]: this no longer points to an existing file. The new file is gmt-4.5.15-src.tar.bz2

File: gmt4.rb [hunk 2]
```diff
@@ -16,8 +16,8 @@ class Gmt4 < Formula
   conflicts_with "gmt", :because => "both versions install the same binaries."
 
   resource "gshhg" do
-    url "ftp://ftp.soest.hawaii.edu/gmt/gshhg-gmt-2.3.4.tar.gz"
```
Comment (cll12) [inline gmt4.rb]: These also have since been updated

File: gmt4.rb (hunk 1 above)
Comment (dpo) [inline gmt4.rb]: The `sha256` must also be updated. That's why the build has failed.

File: gmt4.rb [hunk 3]
```diff
@@ -16,8 +16,8 @@ class Gmt4 < Formula
...
-    mirror "ftp://ftp.star.nesdis.noaa.gov/pub/sod/lsa/gmt/gshhg-gmt-2.3.4.tar.gz"
+    url "ftp://ftp.soest.hawaii.edu/gmt/gshhg-gmt-2.3.6.tar.gz"
+    mirror "ftp://ftp.star.nesdis.noaa.gov/pub/sod/lsa/gmt/gshhg-gmt-2.3.6.tar.gz"
     sha256 "420c6c0df9170015ac0f7c7d472c5a58f8b70a7bf89f162c59dcd70735389110"
```
Comment (dpo) [inline gmt4.rb]: Same here.

File: gmt4.rb (hunk 1 above)
Comment (cll12) [inline gmt4.rb]: How/where would I find out what the new sha256 string be?

Comment (dpo) [inline gmt4.rb]: Run `brew fetch gmt4`. It will tell you what the sha256 of the new archive is.

**Pull request closed without merging**